LibsAndSamples.sln

```diff
@@ -187,6 +187,8 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "IntuneMAMSampleiOS", "tests
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "TrueMam", "TrueMam", "{660867D0-956C-4AF2-8627-03AADF4FCEAA}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.Identity.Test.UIAutomation.Infrastructure", "tests\Microsoft.Identity.Test.Core.UIAutomation\Microsoft.Identity.Test.UIAutomation.Infrastructure.csproj", "{3DC6EC76-D350-4D43-B206-A4CEC8AA36D4}"

```

Comment:
A code walkthough would be good, this is a large PR. 
Consider rebasing and merging some of the commits, to allow folks to review commit by commit.

Reply:
Added code tour. I'll try squashing some commits, but I don't promise anything lol. The changes are small enough, I think code tour should be enough.

Reply:
@bgavrilMS How do you usually do it? Do you reset all changes and then commit with new commits or pick and choose what to squash from existing commits? I've been merging main throughout and squashing may be annoying...